src/main/java/net/snowflake/client/log/JDK14Logger.java

```diff
@@ -50,17 +50,46 @@ public JDK14Logger(String name) {
     String javaLoggingConsoleStdOut =
         System.getProperty(SFSessionProperty.JAVA_LOGGING_CONSOLE_STD_OUT.getPropertyKey());
     if ("true".equalsIgnoreCase(javaLoggingConsoleStdOut)) {
-      useStdOutConsoleHandler();
+      String javaLoggingConsoleStdOutThreshold =
+          System.getProperty(
+              SFSessionProperty.JAVA_LOGGING_CONSOLE_STD_OUT_THRESHOLD.getPropertyKey());
+      useStdOutConsoleHandler(javaLoggingConsoleStdOutThreshold);
     }
   }
 
   @SnowflakeJdbcInternalApi
-  public static void useStdOutConsoleHandler() {
+  public static void useStdOutConsoleHandler(String threshold) {
+    Level thresholdLevel = threshold != null ? tryParse(threshold) : null;
     Logger rootLogger = Logger.getLogger("");
     for (Handler handler : rootLogger.getHandlers()) {
       if (handler instanceof ConsoleHandler) {
         rootLogger.removeHandler(handler);
-        rootLogger.addHandler(STD_OUT_CONSOLE_HANDLER);
+        if (thresholdLevel != null) {
+          rootLogger.addHandler(new StdErrOutThresholdAwareConsoleHandler(thresholdLevel));
+        } else {
+          rootLogger.addHandler(new StdOutConsoleHandler());
+        }
+        break;
+      }
+    }
+  }
+
+  private static Level tryParse(String threshold) {
+    try {
+      return Level.parse(threshold);
+    } catch (Exception e) {
+      throw new UnknownJavaUtilLoggingLevelException(threshold);
+    }
+  }
+
+  @SnowflakeJdbcInternalApi
+  static void resetToDefaultConsoleHandler() {
+    Logger rootLogger = Logger.getLogger("");
+    for (Handler handler : rootLogger.getHandlers()) {
+      if (handler instanceof StdErrOutThresholdAwareConsoleHandler
+          || handler instanceof StdOutConsoleHandler) {
+        rootLogger.removeHandler(handler);
```

Comment:
Should we also explicitly close the handler?

Reply:
I don't think we need to do this - each publish of log writes the message and flushes, close only flushes